### ISSUE TITLE
Endpoint /users/me for accessing private data

### ISSUE THREAD

**Issue by lfaucon** [tournesol-app/tournesol-operations#83]
`/users/:username` will later be used to access public data

**Comment by GresilleSiffle** [comment]
Hello @lfaucon 

Few questions before starting : )

Should I create the route `GET /users/:id/` or `GET /users/:username/`? 

Should the route return all fields present in the [User](https://github.com/tournesol-app/tournesol-backend/blob/main/core/models/user.py) model?

Last one, should we include the [UserPreference](https://github.com/tournesol-app/tournesol-backend/blob/main/core/models/user.py#L201) in the response payload? (and other User's related models or are they out of the scope of this issue?)

Thanks!

**Comment by jstainer-kleis** [comment]
> Should I create the route `GET /users/:id/` or `GET /users/:username/`?

I'd say `GET /users/:username/` since I think we decided that we would avoid exposing IDs from the DB to the outside world (and routes referencing users have been or are rewritten to follow the second pattern).

> Should the route return all fields present in the [User](https://github.com/tournesol-app/tournesol-backend/blob/main/core/models/user.py) model?

I'm not sure if the endpoint should be only used by the user itself (in that case probably almost everything can be returned) or if it will also be used by unauthenticated clients / other users. In the second case I'm not convinced we should use the same endpoint with different responses instead of having separate ones.

> Last one, should we include the [UserPreference](https://github.com/tournesol-app/tournesol-backend/blob/main/core/models/user.py#L201) in the response payload? (and other User's related models or are they out of the scope of this issue?)

I'd say out of scope for now, but for the future I don't know.

**Comment by lfaucon** [comment]
Actually the problem was to get access to username/userId to then be able to make requests, so having this endpoint at ?`/user/:username` will not do it.

I would recommend having a endpoint `/check_auth/` that returns `{id: int, username: str, email: str}` if authenticated of `{msg: 'not authenticated'}` if not. Unless you think this will never be useful? 

UserPreferences or "All fields in the User model" will not yet be included in the subset of Tournesol's features that we plan to deploy asap. Later on they might be served on the endpoint `/users/:username/preferences` and `/users/:username/`
